osgp/protocol-adapter-dlms/osgp-protocol-adapter-dlms/src/main/java/org/opensmartgridplatform/adapter/protocol/dlms/domain/commands/CorrelatedObject.java

```diff
@@ -0,0 +1,40 @@
+package org.opensmartgridplatform.adapter.protocol.dlms.domain.commands;
+
+import java.io.Serializable;
+
+import lombok.Getter;
+import org.opensmartgridplatform.adapter.protocol.dlms.infra.messaging.RequestWithMetadata;
+import org.opensmartgridplatform.dto.valueobjects.smartmetering.ActionRequestDto;
+import org.opensmartgridplatform.shared.infra.jms.MessageMetadata;
+
+@Getter
+public class CorrelatedObject<T> implements ActionRequestDto {
```

Comment:
I find this a bit generic since it adapts ActionRequestDto. Personally I would choose CorrelatedActionRequestDto

Reply:
Maybe even remove the interface ActionRequestDto from CorrelatedObject, extend CorrelatedObject as CorrelatedActionRequest and put the interface on CorrelatedActionRequest. This makes CorrelatedObject usable in other situation where you need a Correlated Object.